python/tank/bootstrap/manager.py

```diff
@@ -574,14 +593,109 @@ def _bootstrap_sgtk(self, engine_name, entity, progress_callback=None):
         else:
             raise TankBootstrapError("Unknown configuration update status!")
 
+        return config, status
+
+    def bootstrap_toolkit(self, entity):
+        """
+        Create an :class:`~sgtk.Sgtk` instance for the given entity.
+
+        If entity is None, the method will bootstrap into the site
+        config. This method will attempt to resolve the configuration and download it
+        locally. Note that it will not cache the application bundles. Use
+        :meth:`ToolkitManager.update_and_cache_configuration` to cache bundles before
+        starting the engine.
+
+        Please note that the API version of the :class:`~sgtk.Sgtk` instance may not be the same as the
+        API version that was used during the bootstrap.
+
+        :param entity: Shotgun entity used to resolve a project context.
+        :type entity: Dictionary with keys ``type`` and ``id``, or ``None`` for the site
+        :param progress_callback: Callback function that reports back on the toolkit bootstrap progress.
+                                  Set to ``None`` to use the default callback function.
+        :returns: Bootstrapped :class:`~sgtk.Sgtk` instance.
+        """
+        # Only return the Toolkit instance.
+        return self._bootstrap_toolkit_internal(entity, None)[0]
+
+    def _bootstrap_toolkit_and_cache_apps(self, engine_name, entity, progress_callback):
+        """
+        Create an :class:`~sgtk.Sgtk` instance for the given entity.
+
+        If entity is None, the method will bootstrap into the site
+        config. This method will attempt to resolve the configuration and download it
+        locally. It will also cache the application bundles.
+
+        :param engine_name: Name of the engine used to find which applications need to be cached.
+        :param entity: Shotgun entity used to resolve a project context.
+        :type entity: Dictionary with keys ``type`` and ``id``, or ``None`` for the site
+        :param progress_callback: Callback function that reports back on the toolkit bootstrap progress.
+                                  Set to ``None`` to use the default callback function.
+        :returns: Bootstrapped :class:`~sgtk.Sgtk` instance.
+        """
+
+        # First get the Toolkit instance
+        tk, status = self._bootstrap_toolkit_internal(entity, progress_callback)
+
+        # Now cache the apps
+        if status != Configuration.LOCAL_CFG_UP_TO_DATE:
+            self._cache_apps(tk, tk.pipeline_configuration, engine_name, entity, progress_callback)
+
+        return tk
+
+    def _bootstrap_toolkit_internal(self, entity, progress_callback):
```

Comment:
The async boostrap calls boostrap_toolkit directly and provides its own progress callback so that it handles thread safety. When the callback is invoke, it sends an event to the main thread which will invoke manager.progress_callback.